File: python/src/problem/leetcode/easy/leetcode_168.py
# coding=utf-8

from src.data_structure.data_structure import ListNode


class Solution:
    """
    Excel 表列名称
    """

    def convert_to_title(self, n: int) -> str:
        """
        Time: O(log_26(n)), Space: O(log_26(n))
        :param n:
        :return:
        """
        s = ""
        while n > 0:
            n -= 1
            s = chr(n % 26 + ord('A')) + s
            n //= 26
        return s

File: java/src/main/java/com/ywh/problem/leetcode/easy/LeetCode168.java
package com.ywh.problem.leetcode.easy;

/**
 * Excel 表格的列标题
 * [数学]
 *
 * @author ywh
 * @since 18/02/2020
 */
public class LeetCode168 {

    /**
     * Time: O(log26(n)), Space: O(log26(n))
     *
     * @param n
     * @return
     */
    public String convertToTitle(int n) {
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            n -= 1;
            char c = (char) (n % 26 + 'A');
            sb.append(c);
            n /= 26;
        }
        return sb.reverse().toString();
    }

}
